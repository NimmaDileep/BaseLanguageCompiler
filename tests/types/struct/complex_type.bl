// Currently fails due to error in array indexing within a struct

struct Message {
    time: Int,
    msg: String,
}

struct Stream {
    name: String,
    messages: [Message],
}

var s: Stream = Stream(
    "YouTube",
    [
        Message(0, "Hello"),
        Message(301, "World"),
    ]
);

print(s.name);
print(s.messages[0].msg);
print(s.messages[1].time);
print(len(s.messages));
