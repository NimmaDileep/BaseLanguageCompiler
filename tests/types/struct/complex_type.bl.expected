YouTube
Hello
301
2
